### ISSUE TITLE
Migrate suite of automated browser tests to our new LambdaTest account

### ISSUE THREAD

**Issue by gabek** [owncast/owncast#3734]
### Share your bug report, feature request, or comment.

LambdaTest is sponsoring us with both unlimited use of their platform, with multiple seats as needed for people working on things, and a monthly donation to the project.

We have been previously using BrowserStack for this functionality.

**Comment by gabek** [comment]
We'll have to replace the BrowserStack logo with the LambdaTest one. LambdaTest suggests the following:
> Please give testing courtesy to LambdaTest in the website footer
and GitHub readme with the below code and send us the screenshot.

```html
<p style="font-size:21px; color:black; ">Browser testing via <a
href="https://www.lambdatest.com/" target="_blank"><img
src="https://www.lambdatest.com/support/img/logo.svg"
style="vertical-align: middle;margin-left:5px" width="147" height="26"
/></a></p>
```

**Comment by mahmed2000** [comment]
Is this just the `/test/automated/screenshots`? I can take a crack at porting the script, but testing if it works is a problem since their API is premium only.

**Comment by gabek** [comment]
I had actually started on this, but for the life of me I can't find the branch :\.

These tests refer to the ones at https://github.com/owncast/owncast/tree/develop/test/automated/browser, where it spins up a browser and verifies certain actions and components exist in different states.

One of the material changes here is moving from the Browser Test tunnel to the [LambdaTest Tunnel](https://www.lambdatest.com/support/docs/sharing-lambda-tunnel/), and [enabling it in the config](https://www.lambdatest.com/support/docs/getting-started-with-cypress-testing/#testing-locally-hosted-or-privately-hosted-projects). So when a test instance of Owncast gets spun up under CI, the automated browser can access it.

But in general, it's the same [Cypress tests](https://www.lambdatest.com/support/docs/getting-started-with-cypress-testing/) that we're already using.

I can create you an account if you'd like to poke around.

As for the automated screenshots, I originally put that together because I thought it would be convenient to always have up to date screenshots to pull from, but I've literally never used them for anything. So when they became flaky it didn't bother me, since they weren't a test, and I wasn't using them. I should just get rid of them, I guess.

<blockquote><img src="https://repository-images.githubusercontent.com/266445736/4f82fb80-a5b1-11ea-8f1c-69d316f861b5" width="48" align="right"><div><img src="https://github.githubassets.com/favicons/favicon.svg" height="14"> GitHub</div><div><strong><a href="https://github.com/owncast/owncast/tree/develop/test/automated/browser">owncast/test/automated/browser at develop · owncast/owncast</a></strong></div><div>Take control over your live stream video by running it yourself.  Streaming + chat out of the box. - owncast/owncast</div></blockquote>

**Comment by mahmed2000** [comment]
Kinda working, kinda not.

Some configs are weirdly not applying in the config file. autostart is necessary (set to false) otherwise it can't work with a pre-initialized tunnel. Could just let it start a new tunnel each time on its own, it would be about the same. Sync is necessary to make it wait. Neither do anything when set in the config file, for some reason, have to set it with each command. Still works once set though.

Mobile tests get rejected. --res is not viewport configuration, the docs say it needs to be done in cypress itself. Can try to look for an env variable or something.

$BUILD_ID acts odd and there's no innate grouping. Its appended by default to the build-name, so have to keep that in mind. There's a separate id that lambdatest uses anyway, so it just seems superfluous.

Branch is here: https://github.com/mahmed2000/owncast/tree/lambdatest-browser-testing

the workflow needs changing and the github integration needs implementation next, I'm assuming?

<blockquote><img src="https://opengraph.githubassets.com/afb03f4740eef118faa9d6b292da289b63d9a0a3947ca2465b5c803937a4abb1/mahmed2000/owncast" width="48" align="right"><div><img src="https://github.githubassets.com/favicons/favicon.svg" height="14"> GitHub</div><div><strong><a href="https://github.com/mahmed2000/owncast">GitHub - mahmed2000/owncast at lambdatest-browser-testing</a></strong></div><div>Take control over your live stream video by running it yourself.  Streaming + chat out of the box. - GitHub - mahmed2000/owncast at lambdatest-browser-testing</div></blockquote>

**Comment by mahmed2000** [comment]
Something about websockets seems kind of iffy with tunnels. Outside of that, tests run fine.

While everything runs, there is no integration for github actions with this testing (the auto-reporting bit). Lambdatest's is with hyperexecute, but if I understand this correctly, that would be the same similar to now, without any tunneling.

I can open a PR for this, if this is as fine as is, and if I can figure out the websocket problem.

**Comment by Ryuno-Ki** [comment]
I noticed that https://www.testmuai.com/resources/images/logo-white.svg responds with a 404 Not Found which ends up on the homepage (repo there doesn't allow for opening issues).

> We'll have to replace the BrowserStack logo with the LambdaTest one. LambdaTest suggests the following:
> 
> > Please give testing courtesy to LambdaTest in the website footer
> > and GitHub readme with the below code and send us the screenshot.
> 
> <p style="font-size:21px; color:black; ">Browser testing via <a
> href="https://www.lambdatest.com/" target="_blank"><img
> src="https://www.lambdatest.com/support/img/logo.svg"
> style="vertical-align: middle;margin-left:5px" width="147" height="26"
> /></a></p>

Is this still relevant? Lambdatest appears to have been bought up by this „AI” thingy.

<blockquote><div><strong><a href="https://www.testmuai.com/resources/images/logo-white.svg">Most Powerful Cross Browser Testing Tool Online | LambdaTest</a></strong></div></blockquote>

**Comment by gabek** [comment]
Yeah, it's probably not so much a thing anymore haha. I should reach out to them and see what's up.

**Comment by Ryuno-Ki** [comment]
FWIW Bulma fixed it like this recently: https://github.com/jgthms/bulma/commit/741da22368cd548111224983a1c3ece8954fe8cc

**Comment by gabek** [comment]
Oh I don't mean I'm concerned about the image, I mean I'm concerned if the new company still has any sponsorship with us at all.
